File: app/src/main/java/org/stepic/droid/web/SearchResultResponse.java
package org.stepic.droid.web;

import com.google.gson.annotations.SerializedName;

import org.stepic.droid.model.Meta;
import org.stepic.droid.model.SearchResult;

import java.util.List;

public class SearchResultResponse extends StepicResponseBase {

    @SerializedName("search-results")
    List<SearchResult> searchResultList;

    public SearchResultResponse(Meta meta, List<SearchResult> searchResultList) {
        super(meta);
        this.searchResultList = searchResultList;
    }

    public List<SearchResult> getSearchResultList() {
        return searchResultList;
    }
}

File: app/src/main/java/org/stepic/droid/web/SearchResultResponse.kt
package org.stepic.droid.web

import com.google.gson.annotations.SerializedName
import org.stepik.android.model.Meta
import org.stepik.android.model.SearchResult

class SearchResultResponse(
        meta: Meta,
        @SerializedName("search-results")
        val searchResultList: List<SearchResult>
) : MetaResponseBase(meta)